scripts/lib/CIME/SystemTests/system_tests_compare_two.py

```diff
@@ -299,7 +301,7 @@ def _get_case2_exeroot(self):
             # to this case. (If case2's CIME_OUTPUT_ROOT were in some
             # more generic location, then this would result in its bld
             # directory being inadvertently shared with other tests.)
-            case2_exeroot = os.path.join(self._get_output_root2(), "bld")
+            case2_exeroot = os.path.join(self._get_output_root2(), self._case1.get_value("CASE"), "bld")
```

Comment:
This was not an inadvertent change - it was changed very intentionally. Please see the above comment.